src/unstract/sdk/x2txt.py

```diff
@@ -68,6 +69,16 @@ def process(
         output_file_path: Optional[str] = None,
         **kwargs: dict[Any, Any],
     ) -> str:
+        if kwargs.get("enable_highlight", False) and isinstance(
+            self._x2text_instance, LLMWhisperer
+        ):
+            output = self._x2text_instance.process_with_hash(
+                input_file_path, output_file_path, **kwargs
+            )
+
+            self._tool.update_exec_metadata({"whisper_hash": output["whisper_hash"]})
```

Comment:
I believe this would create/update a `METADATA.json` in the context of tool execution in a workflow. Do check how this turns out in the context of Prompt Studio

Reply:
In the context of Prompt Studio, we might have to handle it differently. Text extraction is a sub-step while indexing, so we need to record it somewhere. This might require further discussions.

Reply:
This will require taking it up as a different activity, as the current scope is API deployment alone.